programs/bpf_loader/src/syscalls.rs

```diff
@@ -1247,13 +1273,28 @@ impl<'a, 'b> SyscallObject<BpfError> for SyscallKeccak256<'a, 'b> {
                     ),
                     result
                 );
-                let byte_cost = invoke_context.get_compute_budget().sha256_byte_cost;
-                question_mark!(
-                    invoke_context
-                        .get_compute_meter()
-                        .consume(byte_cost * (val.len() as u64 / 2)),
-                    result
-                );
+                if invoke_context
+                    .feature_set
+                    .is_active(&update_syscall_base_costs::id())
+                {
```

Comment:
How about factoring out the common code like this:
```
let cost = if invoke_context.feature_set.is_active(&update_syscall_base_costs::id()) {
    compute_budget.syscall_base_cost.max(compute_budget.sha256_byte_cost.saturating_mul(val.len() as u64 / 2))
} else {
    compute_budget.sha256_byte_cost * (val.len() as u64 / 2)
};
question_mark!(invoke_context.get_compute_meter().consume(cost), result);
```

The same goes for all other changes.